tests/OmniSharp.Roslyn.CSharp.Tests/CompletionFacts.cs

```diff
@@ -0,0 +1,715 @@
+                {
+                    case "Bar":
+                        Assert.True(c.Preselect);
+                        break;
+                    default:
+                        Assert.False(c.Preselect);
+                        break;
+                }
+            });
+        }
+
+        [Theory]
+        [InlineData("dummy.cs")]
+        [InlineData("dummy.csx")]
+        public async Task Returns_members_in_object_initializer_context(string filename)
+        {
+            const string source =
+                @"public class MyClass1 {
+                        public string Foo {get; set;}
+                  }
+
+                    public class MyClass2 {
+
+                        public MyClass2()
+                        {
+                            var c = new MyClass1 {
+                             F$$
+                        }
+                    }
+                ";
+
+            var completions = await FindCompletionsAsync(filename, source);
+            ContainsCompletions(completions.Items.Select(c => c.InsertText), "Foo");
+        }
+
+        [Theory]
+        [InlineData("dummy.cs")]
+        [InlineData("dummy.csx")]
+        public async Task Returns_parameter_name_inside_a_method(string filename)
+        {
+            const string source =
+                @"public class MyClass1 {
+                        public void SayHi(string text) {}
+                  }
+
+                    public class MyClass2 {
+
+                        public MyClass2()
+                        {
+                            var c = new MyClass1();
+                            c.SayHi(te$$
+                        }
+                    }
+                ";
+
+            var completions = await FindCompletionsAsync(filename, source);
+            Assert.Contains(completions.Items, c => c.Label == "text:");
+            Assert.All(completions.Items, c =>
+            {
+                switch (c.Label)
+                {
+                    case "text:":
+                        Assert.True(c.Preselect);
+                        break;
+                    default:
+                        Assert.False(c.Preselect);
+                        break;
+                }
+            });
+        }
+
+        [Theory]
+        [InlineData("dummy.cs")]
+        [InlineData("dummy.csx")]
+        public async Task Returns_declaration_names(string filename)
+        {
+            const string source =
+                @"
+public class MyClass
+{
+    MyClass m$$
+}
+                ";
+
+            var completions = await FindCompletionsAsync(filename, source);
+            Assert.Equal(new[] { "myClass", "my", "@class", "MyClass", "My", "Class", "GetMyClass", "GetMy", "GetClass" },
+                         completions.Items.Select(c => c.Label));
+        }
+
+
+        [Theory]
+        [InlineData("dummy.cs")]
+        [InlineData("dummy.csx")]
+        public async Task Returns_override_signatures(string filename)
```

Comment:
these old override tests were only aiming to cover the rudimentary support we had for overrides in the past.
now with proper usage of completion service it would be  good to extend them

Reply:
I've added several more tests, I'm feeling decent about the coverage now.